src/Evolution/Systems/Cce/KleinGordonSystem.hpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+namespace Cce {
+
+/*!
+ * \brief Performing Cauchy characteristic evolution and Cauchy characteristic
+ * matching for Einstein-Klein-Gordon system.
+ *
+ * \details The code adopts the characteristic formulation to solve the field
+ * equations for scalar-tensor theory as considered in \cite Ma2023sok. Working
+ * in the Einstein frame, a real-valued scalar field \f$\psi\f$ is minimally
+ * coupled with the spacetime metric \f$g_{\mu\nu}\f$. The corresponding action
+ * is expressed as follows:
+ *
+ * \f[
+ * S = \int d^4x \sqrt{-g} \left(\frac{R}{16 \pi}  - \frac{1}{2} \nabla_\mu \psi
+ * \nabla^\mu \psi\right).
+ * \f]
+ *
+ * The system consists of two sectors: scalar and tensor (metric). The scalar
+ * field follows the Klein-Gordon (KG) equation
+ *
+ * \f[
+ * \Box \psi = 0.
+ * \f]
+ *
+ * Its characteristic expression is given in \cite Gomez1996dy, yielding
+ * the hypersurface equation for \f$\partial_u\psi=\Pi\f$, where
+ * \f$\partial_u\f$ represents differentiation with respect to retarded time at
+ * fixed numerical radius \f$y\f$. The code first integrates the KG equation
```

Comment:
Is the fixed numerical radius $y$ or is it $u$? If it's $y$, can you (briefly) introduce how $u$ and $y$ are related?

Reply:
Here $y$ stands for numerical radius and $u$ retarded time. So the sentence had already introduced $y$.

I've added $u$ after `retarded time` to clarify its meaning.